### PR TITLE
Prac 08

### PR DESCRIPTION
**Pull request by SusanMorton**
Hey @Piper-Rains here is my prac 8 code to review. 

I am particularly interested in whether the taxi simulator can be written any more efficiently. Did you use more than 1 extra function in the program? 

Cheers, Susan :)

### CODE REVIEW

File: prac_08/taxi_simulator.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+def main():
+    taxis = [Taxi("Prius", 100), SilverServiceTaxi("Limo", 100, 2), SilverServiceTaxi("Hummer", 200, 4)]
+    current_taxi = None
+    current_bill = 0
+    print("Let's Drive!")
+    print(MENU)
+    menu_choice = input(">>>>> ").lower()
+    while menu_choice != "q":
+        if menu_choice == "c":
```
Comment (Piper-Rains) [inline prac_08/taxi_simulator.py:15]: I added two extra def functions at the end to perform error checking for the user inputs for c and d (just to make sure my program wouldn't crash). Other than that, looks good and efficient.

**Review by Piper-Rains — COMMENTED** [review]
Overall seems good - just add some error checking so it won't crash.

File: prac_08/taxi_simulator.py (hunk 1 above)
Comment (Piper-Rains) [reply prac_08/taxi_simulator.py:15]: So, if the user enters a negative number, something that isn't a number, or tries to select a taxi number that isn't available from the taxi list, the program can identify what has happened, provide feedback to the user and ask them to re-enter their choice.

**Comment by SusanMorton** [comment]
thanks @Piper-Rains :)

**Pull request merged**